Diagnostics/AVTester/Test-ExchAVExclusions.ps1

```diff
@@ -313,6 +314,37 @@ $currentDiff = $initialDiff
 $firstExecution = $true
 $SuspiciousProcessList = New-Object Collections.Generic.List[string]
 $SuspiciousW3wpProcessList = New-Object Collections.Generic.List[string]
+$SuspiciousAMSIinW3wpProcessList = New-Object Collections.Generic.List[string]
+
+# Get AMSI Dlls registered
+# Define the AMSI providers registry path
+$registryPath = "HKLM:\SOFTWARE\Microsoft\AMSI\Providers"
+
+# Get all subKeys in the specified registry path
+$subKeys = Get-ChildItem -Path $registryPath -ErrorAction SilentlyContinue
+$AMSIDlls = New-Object Collections.Generic.List[string]
+
+if ($subKeys) {
+    # Regular expression to match the subKey names
+    $regex = "[0-9A-Fa-f\-]{36}"
+
+    # Filter the subKeys that match the regular expression and get only their names
+    $matchingSubKeys = $subKeys | Where-Object { $_.Name -match $regex } | ForEach-Object { $_.PSChildName }
+
+    if ($matchingSubKeys) {
+        $AMSIDll = $null
+        $AMSIDll = $matchingSubKeys | ForEach-Object { (Get-Item "HKLM:\SOFTWARE\Classes\ClSid\$_\InprocServer32" -ErrorAction SilentlyContinue).GetValue("") }
+        if ($AMSIDll) {
+            $AMSIDlls += $AMSIDll.trim('"')
```

Comment:
`$AMSIDlls` was created as a generic list, but you are using it as a static array causing it to switch. Please use `$AMSIDlls.Add(($AMSIDll.trim('"')))` instead.

Reply:
if fact it is not need it, AMSIDll has the information need it